src/p2p/ServiceQueue.ts

```diff
@@ -727,6 +823,96 @@ function sortedInsert(
   }
 }
 
+function verifyDebugDropNGT(reqParamsDropNGT, cycle): { success: boolean; message: string } {
+  const payload = {
+    route: stripQueryParams(reqParamsDropNGT.url, ['sig', 'sig_counter', 'nodePubkeys']), //<- we're gonna hash, these query artificats need to be excluded from the hash
+    count: reqParamsDropNGT.sigCounter,
+    nodes: reqParamsDropNGT.pubKeys,
+    networkId: cycle.networkId,
+    cycleCounter: cycle.counter,
+  }
+  const hash = crypto.hash(Utils.safeStringify(payload))
+  const devPublicKeys = getDevPublicKeys() // This should return list of public keys
+  const requestSig = reqParamsDropNGT.sig
+
+  if (reqParamsDropNGT.owner !== '') {
```

Comment:
looks like logic in this condition and loop below is exactly the same... if so, could you move this in a method instead of duplicating the logic?

Reply:
ok